Lesson26/app/deniz/script.js

```diff
@@ -0,0 +1,52 @@
+// Register on Unsplash to get your own API key
+const API_KEY = '3HO6COGJ_0Qsi2DxSVvlppxQBAHZj4jCPBlARAusM3g';
+
+// TODO: Add a new button in HTML to crear the screen
+
+const button = document.getElementById('fetch-button');
+button.addEventListener('click', function () {
+  let picCount = parseInt(document.getElementById("pic-count").value) || 10;
+  
+  if (picCount <= 0) {
+    document.getElementById("error-message").textContent = "Please enter a postive number.";
+    return;
+  }
+  // TODO: Add an input in HTML (e.g. slider) to select how many new picutres should be fetched. Depending on the user input, the count value in the request should be changed to value selected by user
+  fetch(`https://api.unsplash.com/photos/random?count=${picCount}&client_id=${API_KEY}`)
+    .then(response => {
+      if (!response.ok) {
+        throw new Error("Network response was not ok " + response.statusText);
+      }
+      return response.json();
+    })
+    .then(data => {
+      if (data.errors) {
+        throw new Error(data.errors[0]); // burada neden (data.errors[0]) kullaniyoruz ?
+      }
+      const imageContainer = document.getElementById("image-container");
+
+      // TODO: Add error handling (display error message to the user if an error occurs)
+      data.map(imageData => {
+        const imageWrapper = document.createElement("div");
+        imageWrapper.classList.add('image-wrapper');
+        const imageElement = document.createElement("img");
+
+        imageElement.src = imageData.urls.small;
+        imageWrapper.append(imageElement);
+        document.getElementById("image-container").append(imageWrapper);
+      })
+    })
+    .catch(error => {
+      console.error('There was a problem with the fetch operation:', error);
+      alert('Error fetching images. Please try again later.');
```

Comment:
Avoid using alert function, it's considered bad user experience. Instead, add some text to the 'error-message' div if you need to inform a user about the error.